### DISCUSSION TITLE
What's the purpose of the obfuscated file (init_patch.ts)?

### DISCUSSION THREAD

**Discussion by renandecarlo** [open-wa/wa-automate-nodejs#2809] [Q&A]
I know wa-automate-nodejs requires a license for some of its functionality.
However, it also says it blocks messages from unknown numbers, and sends a notification back to host so they know it's being used by automation software.

None of those last 2 are working. It doesn't send the notification message, and it also allows messages to unknown numbers. So, what's the purpose of the obfuscated file?

**Comment by smashah** [comment]
Host notifications have been disabled for now.

The init patch makes sure everything is set up for the patches to actually work.

` it also allows messages to unknown numbers` Are you sure about this? This shouldn't be possible without a license.

In order to check, use [`getContact`](https://docs.openwa.dev/classes/api_Client.Client.html#getContact)

if it returns an object then it is technically a contact of your host account (even though it might not be in your addressbook)

The ability to send message to non-contacts is locked down to get businesses to pay for a license key. If for some reason this functionality is open then I would expect business users to be adults and actually still get a license key.

**Reply by renandecarlo** [reply]
I see. 

> ` it also allows messages to unknown numbers` Are you sure about this? This shouldn't be possible without a license.

It did for me on a phone without any contacts. 
It also "sent" messages to non existing numbers. As in the message went through, as if the contact was a valid WPP one, but it was never received.
